### PR TITLE
added test for removing invalid lines with DROPMALFORMED

### PR DESCRIPTION
**Pull request by kmader**
Added a test file and the accompanying tests (which currently fail)

### CODE REVIEW

**Comment by kmader** [comment]
For me to compile it, I needed to use maven (https://gist.github.com/kmader/f046235024f375e9d81d) not sure why the SBT does not work. My configuration is pretty standard, but even the Readme reports a broken build.

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala [hunk 1]
```diff
@@ -271,6 +272,45 @@ abstract class AbstractCsvSuite extends FunSuite with BeforeAndAfterAll {
     assert(results === 0)
   }
 
+  test("DSL test with poorly formatted file and string schema") {
+    val stringSchema = new StructType(
+      Array(
+        StructField("Name",StringType,true),
```
Comment (falaki) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: Style nits:
```
StructField("Name", StringType, true),
StructField("Age", StringType, true),
StructField("Height", StringType, true)
```

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala [hunk 2]
```diff
@@ -271,6 +272,45 @@ abstract class AbstractCsvSuite extends FunSuite with BeforeAndAfterAll {
...
+      .withParseMode("DROPMALFORMED")
+      .csvFile(sqlContext, ageFile)
+      .count()
+
+    assert(results === 3)
+  }
+  test("DSL test with poorly formatted file and known schema") {
+    val strictSchema = new StructType(
+      Array(
+        StructField("Name",StringType,true),
```
Comment (falaki) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: Style nits:
```
StructField("Name", StringType, true),
StructField("Age", Integer, true),
StructField("Height", DoubleType, true)
```

**Comment by falaki** [comment]
@kmader Thanks for submitting this. I left two minor style comments, otherwise LGTM.

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 3]
```diff
@@ -123,6 +123,10 @@ case class CsvRelation protected[spark] (
           case aiob: ArrayIndexOutOfBoundsException if permissive =>
             (index until schemaFields.length).foreach(ind => rowArray(ind) = null)
             Some(Row.fromSeq(rowArray))
+          case nfe: java.lang.NumberFormatException if dropMalformed =>
```
Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: Just realized we are silently dropping data here. Please add a warning like this:
```
logger.warn(s"Number format error. Dropping malformed line: ${tokens.mkString(",")}")
```

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 4]
```diff
@@ -123,6 +123,10 @@ case class CsvRelation protected[spark] (
...
+            None
+          case pe: java.text.ParseException if dropMalformed =>
```
Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: Same here.

File: src/main/scala/com/databricks/spark/csv/CsvRelation.scala [hunk 5]
```diff
@@ -123,6 +123,14 @@ case class CsvRelation protected[spark] (
           case aiob: ArrayIndexOutOfBoundsException if permissive =>
             (index until schemaFields.length).foreach(ind => rowArray(ind) = null)
             Some(Row.fromSeq(rowArray))
+          case nfe: java.lang.NumberFormatException if dropMalformed =>
+            logger.warn("Number format exception. "+
+              s"Dropping malformed line: ${tokens.mkString(",")}")
+            None
+          case pe: java.text.ParseException if dropMalformed =>
+            logger.warn("Parse Exception. "+
```
Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: The style checker is complaining about this line:
```
logger.warn("Parse exception. " + ...
```

Comment (kmader) [inline src/main/scala/com/databricks/spark/csv/CsvRelation.scala]: sorry, I had disabled that on my setup

**Pull request closed without merging**